extensions/wikia/CategoryPage3/CategoryPage3.class.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+use Wikia\Template\Engine;
+use Wikia\Template\PHPEngine;
+
+class CategoryPage3 extends CategoryPageWithLayoutSelector {
+	/**
+	 * @var String - query param used for pagination
+	 */
+	private $from;
+
+	/**
+	 * @var CategoryPage3Model
+	 */
+	private $model;
+
+	/**
+	 * @var array
+	 */
+	private $trendingPages;
+
+	public function openShowCategory() {
+		parent::openShowCategory();
+
+		// Use AssetsManager for styles because it bundles all styles and blocks render so there is no FOUC
+		\Wikia::addAssetsToOutput( 'category_page3_scss' );
+	}
+
+	/**
+	 * @throws Exception
+	 */
+	public function closeShowCategory() {
+		$context = $this->getContext();
+		$request = $context->getRequest();
+		$this->from = $request->getVal( 'from' );
+
+		$this->model = new CategoryPage3Model( $context->getTitle(), $this->from );
+		$this->model->loadData();
+
+		if ( $this->model->getTotalNumberOfMembers() === 0 ) {
+			return;
+		}
+
+		$this->model->loadImages( 40, 30 );
```

Comment:
please move these number to some consts. It will be easier to read the code, as `40, 30` doesn't tell me anything

Reply:
![screenshot from 2018-11-02 13-27-12](https://user-images.githubusercontent.com/7030884/47915432-236a3a80-dea3-11e8-9c80-c79827f2021f.png)
Isn't that redundant? ;)